examples/nextjs/hooks/sample-app/src/app/layout.tsx

```diff
@@ -1,22 +1,54 @@
-import './globals.css'
-import type { Metadata } from 'next'
-import { Inter } from 'next/font/google'
+import './globals.css';
+import type { Metadata } from 'next';
+import { Inter } from 'next/font/google';
+import Script from 'next/script';
 
-const inter = Inter({ subsets: ['latin'] })
+const inter = Inter({ subsets: ['latin'] });
 
 export const metadata: Metadata = {
   title: 'Create Next App',
   description: 'Generated by create next app',
-}
+};
 
-export default function RootLayout({
-  children,
-}: {
-  children: React.ReactNode
-}) {
+export default function RootLayout({ children }: { children: React.ReactNode }) {
   return (
-    <html lang="en">
+    <html lang='en'>
+      <head>
+        <Script id='bufferEvents'>
+          {`
+            window.rudderanalytics = [];
+            var methods = [
+              'load',
+              'page',
+              'track',
+              'identify',
+              'alias',
+              'group',
+              'ready',
+              'reset',
+              'getAnonymousId',
+              'setAnonymousId',
+              'getUserId',
+              'getUserTraits',
+              'getGroupId',
+              'getGroupTraits',
+              'startSession',
+              'endSession',
+              'getSessionId',
+            ];
+            for (var i = 0; i < methods.length; i++) {
+              var method = methods[i];
+              window.rudderanalytics[method] = (function (methodName) {
+                return function () {
+                  window.rudderanalytics.push([methodName].concat(Array.prototype.slice.call(arguments)));
+                };
+              })(method);
+            }
+            window.rudderanalytics.page('sample page call');
```

Comment:
can we adda comment above the page call please that this is only for demostration reason, if auto page call needs to take place its better to do in the spa code

Reply:
Addressed